lib/routemaster/errors.rb

```diff
@@ -0,0 +1,73 @@
+module Routemaster
+  module Errors
+    class BaseError < RuntimeError
+      attr_reader :env
+
+      def initialize(env)
+        @env = env
+        super(message)
+      end
+
+      def errors
+        body['errors']
+      end
+
+      def message
+        raise NotImplementedError
+      end
+
+      def body
+        @body ||= deserialize(@env.body)
+      end
```

Comment:
we don't actually need to pass `@env.body` to the `deserialize` method. It has access to it as well.